pepper-apis/src/main/java/org/broadinstitute/ddp/event/dsmtask/api/DsmTaskPubSubConnectionCreator.java

```diff
@@ -0,0 +1,110 @@
+ * After doing an action it is called the {@link DsmTaskResultSender}
+ * which sends result to topic "dss-to-dsm-results".
+ */
+public class DsmTaskPubSubConnectionCreator {
+
+    private static final Logger LOG = getLogger(DsmTaskPubSubConnectionCreator.class);
+
+    private static final long SUBSCRIBER_AWAIT_RUNNING_TIMEOUT = 30L;
+
+    private final PubSubConnectionManager pubSubConnectionManager;
+    private final ProjectSubscriptionName projectSubscriptionName;
+    private final ProjectTopicName dsmTaskResultProjectTopicName;
+    private final DsmTaskProcessorFactory dsmTaskProcessorFactory;
+
+    private DsmTaskReceiver dsmTaskReceiver;
+    private DsmTaskResultSender dsmTaskResultSender;
+
+    private Subscriber dsmTaskSubscriber;
+    private Publisher dsmTaskResultPublisher;
+
+
+    public DsmTaskPubSubConnectionCreator(
+            PubSubConnectionManager pubSubConnectionManager,
+            String projectId,
+            String subscription,
+            String dsmTaskResultPubSubTopic,
+            DsmTaskProcessorFactory dsmTaskProcessorFactory) {
+        this.pubSubConnectionManager = pubSubConnectionManager;
+        projectSubscriptionName = ProjectSubscriptionName.of(projectId, subscription);
+        dsmTaskResultProjectTopicName = ProjectTopicName.of(projectId, dsmTaskResultPubSubTopic);
+        this.dsmTaskProcessorFactory = dsmTaskProcessorFactory;
+    }
+
+    public void create() {
+        try {
+            createDsmTaskResultPubSubTopicPublisher();
+            dsmTaskResultSender = new DsmTaskResultSender(dsmTaskResultPublisher);
+            dsmTaskReceiver = new DsmTaskReceiver(projectSubscriptionName, dsmTaskProcessorFactory, dsmTaskResultSender);
+            createDsmTaskPubSubTopicSubscriber();
+        } catch (Exception e) {
+            LOG.error(errorMsg("Failed to create DsmTask pubsub connection"), e);
+        }
+    }
+
+    public void destroy() {
+        if (dsmTaskSubscriber != null) {
+            dsmTaskSubscriber.stopAsync();
+        }
+        if (dsmTaskResultPublisher != null) {
+            try {
+                dsmTaskResultPublisher.shutdown();
+                dsmTaskResultPublisher.awaitTermination(1, TimeUnit.MINUTES);
+            } catch (Exception e) {
+                LOG.error(errorMsg("Failed to shutdown DsmTask pubsub connection"), e);
+            }
+        }
+    }
+
+    private void createDsmTaskPubSubTopicSubscriber() {
+        dsmTaskSubscriber = pubSubConnectionManager.subscribeBuilder(projectSubscriptionName, dsmTaskReceiver)
+                .setParallelPullCount(1)
+                .setExecutorProvider(InstantiatingExecutorProvider.newBuilder()
+                        .setExecutorThreadCount(1)
+                        .build())
+                .build();
+        try {
+            dsmTaskSubscriber.startAsync().awaitRunning(SUBSCRIBER_AWAIT_RUNNING_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
Please add a handler that responds to subscription errors by re-subscribing, [as outlined here](https://cloud.google.com/pubsub/docs/samples/pubsub-subscriber-error-listener).

Reply:
Added